concent_api/core/tests/test_unit_validation.py

```diff
@@ -268,3 +271,50 @@ def test_that_if_frames_are_not_one_after_the_other_method_should_pass(self):
     def test_that_if_frames_are_not_integers_method_should_raise_exception(self):
         with self.assertRaises(FrameNumberValidationError):
             validate_frames(['1', '2'])
+
+
+class TestValidateComputeTaskDef(object):
+    compute_task_def = None
+
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        self.compute_task_def = ComputeTaskDefFactory()
+        self.compute_task_def["extra_data"] = {
+            "output_format": "PNG",
+            "scene_file": "nice_photo.blend",
+            "frames": [1, 2, 3],
+        }
+
+    def test_that_mising_extra_data_causes_message_validation_error(self):
+        del self.compute_task_def['extra_data']
+        with pytest.raises(MessageValidationError) as e:
+            validate_compute_task_def(self.compute_task_def)
+        assert_that(e.value.error_message).contains(f"extra_data")
+        assert_that(e.value.error_code).is_equal_to(ErrorCode.MESSAGE_INVALID)
+
+    @pytest.mark.parametrize(
+        "missing_data", [
+            "output_format",
+            "scene_file",
+            "frames",
+        ]
+    )
+    def test_that_missing_entries_in_extra_data_causes_message_validation_error(self, missing_data):
+        del self.compute_task_def["extra_data"][missing_data]
+        with pytest.raises(MessageValidationError) as e:
+            validate_compute_task_def(self.compute_task_def)
+        assert_that(e.value.error_message).contains(f"{missing_data}")
+        assert_that(e.value.error_code).is_equal_to(ErrorCode.MESSAGE_INVALID)
+
+    @pytest.mark.parametrize(
+        "value_with_wrong_type", [
+            "output_format",
+            "scene_file",
+        ]
+    )
+    def test_that_wrong_field_types_causes_message_validation_error(self, value_with_wrong_type):
+        self.compute_task_def["extra_data"][value_with_wrong_type] = mock.sentinel.wrongtype
```

Comment:
Could you explain what `mock.sentinel.wrongtype` does ? I was trying to find some docs for it over the web but I couldn't. Looks like a cool feature.

Reply:
https://docs.python.org/3/library/unittest.mock.html#unittest.mock.sentinel
Sentinels are dummy objects (~objects with no value).
Basically, I would say you can use them in two scenarios:
1. For testing that if some function has been feed with certain object, the exact object is returned. For example:
```python
d = { "a": 1, "b": 2}
def get_ value_from_alphabetically_lowest_key(d):
    ....
    return d[key]

assert id(d["a"]) ==  id(value_from_alphabetically_lowest_key(d))
```
2. For testing with non-existing value (my use case) - when we want to check how something behaves when wrong value is given